### PR TITLE
OKx: Fix websocket subscriptions and casing

### PR DESCRIPTION
**Pull request by gloriousCode**
# PR Description
G K's profile says he will be away for a bit, and #1202 has not had an answer. This PR shamelessly steals the work from #1202  and also applies a fix for instrument type casing allowing "orders" auth subscription to work:

```
[DEBUG] | WEBSOCKET | 08/06/2023 13:22:19 | Okx websocket connection: message received: {"event":"subscribe","arg":{"channel":"orders","instType":"FUTURES"}}
```

## Type of change
- [x] Bug fix (non-breaking change which fixes an issue)

## How has this been tested
I tested #1202 and then also tested that now things subscribe properly

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation and regenerated documentation via the documentation tool
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally and on Github Actions/AppVeyor with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by shazbert — APPROVED** [review]
tACK. Thanks.
```
{"event":"subscribe","arg":{"channel":"trades","instId":"KISHU-USDT"}} 💋 U 2
```

**Review by thrasher- — APPROVED** [review]
tACK, thanks for making those changes

File: exchanges/okx/okx_websocket.go [hunk 1]
```diff
@@ -1232,38 +1232,41 @@ func (ok *Okx) wsProcessTickers(data []byte) error {
 func (ok *Okx) GenerateDefaultSubscriptions() ([]stream.ChannelSubscription, error) {
 	var subscriptions []stream.ChannelSubscription
 	assets := ok.GetAssetTypes(true)
+	subs := make([]string, len(defaultSubscribedChannels))
+	copy(subs, defaultSubscribedChannels)
 	if ok.Websocket.CanUseAuthenticatedEndpoints() {
-		defaultSubscribedChannels = append(defaultSubscribedChannels,
+		subs = append(subs,
```
Comment (thrasher-) [inline exchanges/okx/okx_websocket.go:1238]: Just noticed this: https://github.com/thrasher-corp/gocryptotrader/actions/runs/5216358886/jobs/9414995924?pr=1221#step:4:2832

File: exchanges/okx/okx_websocket.go [hunk 2]
```diff
@@ -1232,38 +1232,41 @@ func (ok *Okx) wsProcessTickers(data []byte) error {
 func (ok *Okx) GenerateDefaultSubscriptions() ([]stream.ChannelSubscription, error) {
 	var subscriptions []stream.ChannelSubscription
 	assets := ok.GetAssetTypes(true)
+	var subs []string
+	copy(subs, defaultSubscribedChannels)
```
Comment (thrasher-) [inline exchanges/okx/okx_websocket.go:1236]: This still isn't correct as copy doesn't allocate memory resulting in an empty subs, you can instead do something like this:

```
subs := make([]string, 0, len(defaultSubscribedChannels) + 2)
subs = append(subs, defaultSubscribedChannels...)
if ok.Websocket.CanUseAuthenticatedEndpoints() {
    subs = append(subs, okxChannelAccount, okxChannelOrders)
}
```

**Review by thrasher- — APPROVED** [review]
tACK, thanks for making those changes!

**Review by shazbert — APPROVED** [review]
tested ACK.

**Pull request merged**